src/java.base/unix/classes/java/lang/ProcessImpl.java

```diff
@@ -69,6 +69,10 @@ final class ProcessImpl extends Process {
     // Linux platforms support a normal (non-forcible) kill signal.
     static final boolean SUPPORTS_NORMAL_TERMINATION = true;
 
+    // Cache for JNU Charset. The encoding name is guaranteed
+    // to be supported in this environment.
+    static final Charset JNU_CHARSET = Charset.forName(StaticProperty.jnuEncoding());
```

Comment:
Before the change in this PR, the `jnuCharset` used to fallback to `Charset.defaultCharset()`, if the `sun.jnu.encoding` value wasn't a legal charset or couldn't be loaded. 
On this line, we no longer do that. The comment, above this line, indicates that this is intentional. Looking at the code in `java.lang.System#initPhase1()`, the `sun.jnu.encoding` property value is updated to `UTF-8` if the previous value isn't a supported charset. That is done just before the `StaticProperty` class is referenced and loaded. Effectively, as noted in this comment, the `sun.jnu.encoding` would just have the right usable/supported value stored in `StaticProperty.jnuEncoding()` by the time we reach here. So this change looks fine to me.